src/plugins/embeddable/public/bootstrap.ts

```diff
@@ -17,55 +17,39 @@
  * under the License.
  */
 
-import { UiActionsSetup, Trigger } from 'src/plugins/ui_actions/public';
+import { UiActionsSetup } from 'src/plugins/ui_actions/public';
 import {
-  CONTEXT_MENU_TRIGGER,
-  APPLY_FILTER_TRIGGER,
+  applyFilterTrigger,
+  contextMenuTrigger,
   createFilterAction,
-  PANEL_BADGE_TRIGGER,
-  SELECT_RANGE_TRIGGER,
-  VALUE_CLICK_TRIGGER,
+  panelBadgeTrigger,
+  selectRangeTrigger,
+  valueClickTrigger,
+  EmbeddableVisTriggerContext,
 } from './lib';
 
+declare module '../../ui_actions/public' {
+  export interface TriggerContextMapping {
+    SELECT_RANGE_TRIGGER: EmbeddableVisTriggerContext;
+    VALUE_CLICK_TRIGGER: EmbeddableVisTriggerContext;
+    CONTEXT_MENU_TRIGGER: object;
+    APPLY_FILTER_TRIGGER: object;
+    PANEL_BADGE_TRIGGER: object;
```

Comment:
These should reference the actual ids, there is actually a mismatch right now for the apply filter one because the string is actually "FILTER_TRIGGER".



```
  export interface TriggerContextMapping {
    [SELECT_RANGE_TRIGGER]: EmbeddableVisTriggerContext;
    [VALUE_CLICK_TRIGGER]: EmbeddableVisTriggerContext;
    [APPLY_FILTER_TRIGGER]: {
      embeddable: IEmbeddable;
      filters: Filter[];
    };
    [CONTEXT_MENU_TRIGGER]: object;
    [PANEL_BADGE_TRIGGER]: object;
  }
```

Reply:
Not in this PR, but it'd probably be perferable for those other types to be mapped to undefined or null rather than an object or it won't complain if you _do_ pass data along.

Reply:
I see you fixed the `FILTER_TRIGGER` problem.

> Not in this PR, but it'd probably be perferable for those other types to be mapped ...

Yes, we should type it to the correct type.